pkg/kontainer-engine/drivers/util/utils.go

```diff
@@ -32,13 +32,20 @@ func UserAgentForCluster(cluster *managementv3.Cluster) string {
 
 // GenerateServiceAccountToken generate a serviceAccountToken for clusterAdmin given a rest clientset
 func GenerateServiceAccountToken(clientset kubernetes.Interface, clusterName string) (string, error) {
-	_, err := clientset.CoreV1().Namespaces().Create(context.TODO(), &v1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: cattleNamespace,
-		},
-	}, metav1.CreateOptions{})
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return "", err
+	// intentionally first did a get call to check if the namespace already exist
+	// since in case of etcd-restore operation apiserver tries to verify the create namespace call (due to ValidatingWebhookConfiguration)
+	// and the request can fail based on rancher-webhook pod's running status during restore.
+	_, err := clientset.CoreV1().Namespaces().Get(context.TODO(), cattleNamespace, metav1.GetOptions{})
+	if err != nil {
+		_, err = clientset.CoreV1().Namespaces().Create(context.TODO(), &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cattleNamespace,
+			},
+		}, metav1.CreateOptions{})
+		// checking for already exist error to handle case of a timeout error of the get request
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return "", fmt.Errorf("error creating namespace: %v", err)
+		}
```

Comment:
```suggestion
	_, err := clientset.CoreV1().Namespaces().Get(context.TODO(), cattleNamespace, metav1.GetOptions{})
	if err != nil {
		if !errors.IsNotFound(err) {
			return err
		}
		_, err = clientset.CoreV1().Namespaces().Create(context.TODO(), &v1.Namespace{
			ObjectMeta: metav1.ObjectMeta{
				Name: cattleNamespace,
			},
		}, metav1.CreateOptions{})
		// checking for already exist error to handle case of a timeout error of the get request
		if err != nil && !errors.IsAlreadyExists(err) {
			return "", fmt.Errorf("error creating namespace: %v", err)
		}
```
Although the webhook running may affect the create call, we should be able to tell if it doesn't exist with GET, unless there is some other situation where it may return an error other than `IsNotFound` but a create would succeed that I'm not aware about in this case.

Reply:
done..updated the code..added the NotFound error check and also removed the already exist error check while namespace creation